### ISSUE TITLE
Infection is broken

### ISSUE THREAD

**Issue by pushmain** [CleverRaven/Cataclysm-DDA#47041]
# Describe the bug
Infection is a game breaker now 

# Steps To Reproduce

1. Get the infection
2. Check fatigue buildup, speed penalty. After 3 hours the status is Exhausted, no Stamina recover.
Almost no actions can be performed.
3. Check thirst level.  Drinks consumption required is more than 3 liters of liquid per day. 
Observed consumption of 61 portions in 52 hours (3.5 days)
Check Infection Effect description: Rare:Thirst.

Makes Challenge - Really Bad Day Scenario extremely challenging.
# Expected behavior

Should revert #46795
It was an attempt to fix low body temperature.
However, low body temperature is a common Sepsis sign, see https://en.wikipedia.org/wiki/Systemic_inflammatory_response_syndrome, it can be lower then 36 C / 96.8 F.

It takes about 6 hours to feel incredibly sick, not just 2-3 hours. 
If untreated, in some cases it can kill in 12 hours.
There is a 10% increase in mortality for every hour where treatment with antibiotics is delayed.
See statistics https://www.irishexaminer.com/news/arid-30824690.html
According to other source, however, its only 4% increase
https://invisiverse.wonderhowto.com/news/every-hour-untreated-sepsis-survival-drops-4-here-are-symptoms-watch-for-0177746/

# Versions and configuration

build 0.E-11401b

**Comment by ZhilkinSerg** [comment]
> Makes Challenge - Really Bad Day Scenario extremely challenging.

That's okay.

**Comment by pushmain** [comment]
The cause is not okay. No infection incapacitate in 3 hours.

**Comment by Zireael07** [comment]
> No infection incapacitate in 3 hours

Quote from Spanish flu wiki: his more serious type would cause heliotrope cyanosis to develop, whereby the skin would first develop two mahogany spots over the cheekbones which would then over a few hours spread to color the entire face blue, followed by black coloration first in the extremities and then further spreading to the limbs and the torso. After this, death would follow within hours or days due to the lungs being filled with fluids

Also severe pain can incapacitate very quickly...

**Comment by ZhilkinSerg** [comment]
> 
> 
> The cause is not okay. No infection incapacitate in 3 hours.

First thing, challenges are meant to be challenging. Then you don't know specifics of infection too: what exact infection it is, how long since it settled in your body, etc.

**Comment by pushmain** [comment]
> After this, death would follow within hours or days due to the lungs being filled with fluids

There is a difference between hours and days. Have to stick to average values. 

> Also severe pain can incapacitate very quickly

No pain from infection in the game now.

> First thing, challenges are meant to be challenging

Agree, no question with it. But it should not boil down to playing purely against RNG, when if you not found antibiotics in first 5 houses, you are dead.
I had hundreds of runs of this scenario. It sometimes took all day to find antibiotics.

> Then you don't know specifics of infection too: what exact infection it is, how long since it settled in your body, etc.

I think it should depends how much fun it should produce.
As it is now, the scenario starts in the terminal phase of infection.

**Comment by anothersimulacrum** [comment]
CC @Venera3

**Comment by Venera3** [comment]
The infection as currently implemented is a soft tissue infection with secondary generalization, with the "bite" effect representing the local process. Those are up there on the severity totem pole, since they usually involve multiple species and have a reservoir in the wound itself that the immune system obviously can't clear on its own. Ideally you'd need to clean the wound to have any chance of survival at all, but revamping them wholesale is a project  for later.

Really Bad Day starts you in a fail state, and it's not coy about it. If you expect a junkie with withdrawal, the flu and sepsis to have a realistic chance of survival in the apocalypse you *severely* underestimate all three of those things. That combination would have about a 50/50 chance with the standard of care, if we're being honest.

With that out of the way, with the infected starts you have about until midday to find a place to crash before fatigue makes that choice for you, with the first two hour or so being relatively fit. Afterwards, you take a little catnap until 8-9 PM, where you have about an hour again to take care of water and (hopefully) take your antibiotics. The thirst messages are scary, but they won't do any more than sap your speed. Stamina regen is positive on average until the last stage, when it falls to zero.

**Comment by pushmain** [comment]
> That combination would have about a 50/50 chance with the standard of care, if we're being honest.

I have not seen any data confirming 50/50 chance. I think provided mortality statistic covers wide range of cases. I believe regular healthy person does not get bad cases of infections much.

> Afterwards, you take a little catnap until 8-9 PM, where you have about an hour again to take care of water and (hopefully) take your antibiotics.

I have found that resting does not regenerate stamina at all after some point.  Sleep would only take more time.

Scenario aside, 2-3 hours of time that the commit aims to, is not realistic and not nearly enough in game.
Basically, you have be prepared to survive the infection. Unless strong antibiotics and plenty of water are stockpiled, chances to survive is nonexistent.

**Comment by actual-nh** [comment]
> > That combination would have about a 50/50 chance with the standard of care, if we're being honest.
> 
> I have not seen any data confirming 50/50 chance. I think provided mortality statistic covers wide range of cases. I believe regular healthy person does not get bad cases of infections much.

Possibly. This is not, however, a "regular healthy person" - note "combination". (And, actually, there are ones that are potentially worse - MRSA (or even worse, VRSA) with necrotizing fasciitis ("flesh-eating bacteria") would not, realistically, be taken care of by any antibiotics you're going to find in houses... I've had a quite healthy student who nearly lost a leg from that one.)

**Comment by pushmain** [comment]
> This is not, however, a "regular healthy person" - note "combination".
Lets drop the scenario case. Its just the only case I have tested.
The commit offer 2-3 hours for a otherwise healthy character. Please see the description.

**Comment by actual-nh** [comment]
> > This is not, however, a "regular healthy person" - note "combination".
> 
> Lets drop the scenario case. Its just the only case I have tested.
> The commit offer 2-3 hours for an otherwise healthy character. Please see the description.

2-3 hours while completely functional; the character should not immediately go to completely nonfunctional right after that. Till about midday means a total of ~12 hours before the character crashes, IIRC.

**Comment by pushmain** [comment]
> I suspect the Stamina problem is at least partially caused by the other problems from the scenario.

Unlikely, because

> The fatigue changes alone ensure that an infected character has about 2-3 hours of activity tops in their day

**Comment by actual-nh** [comment]
> > I suspect the Stamina problem is at least partially caused by the other problems from the scenario.
> 
> Unlikely, because
> 
> > The fatigue changes alone ensure that an infected character has about 2-3 hours of activity tops in their day
> 
> Let alone the dehydration

Hrm. The line starting "The fatigue changes alone" is from the PR. I am now somewhat confused.

**Comment by Venera3** [comment]
We have way too many words denoting how knackered the survivor gets. "Fatigue" is the way "sleepiness" is referred to internally, and was the cause of the whole limbs-freezing-off thing. Sorry for the confusion.

>I believe regular healthy person does not get bad cases of infections much.

Regular people go to the hospital with their bites, get them surgically cleaned, get iv broad-specturm antibiotics at the first sign of trouble and *still* develop life-threatening infections. Mortality data in complicated soft tissue infections is a tad harder to track down compared to general sepsis mortality (already not great), but most studies land on ~25% with such fancy-schmancy things as "modern medicine", "ICUs" and "clean ORs". 
Ex.
[[ 1 ]]( https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3782925/ ) - a relatively general overview;
[[ 2 ]](https://jamanetwork.com/journals/jamasurgery/article-abstract/508376) - this one brings up iv drug usage, with a ~4 odds ratio on both overall mortality and amputations.

I did the testing on an Infected start, and splatted a lot. But the timeline was enabling enough to get two survivors to find antibiotics ( yeeting a car into a doc's office and a pharmacy, respectively). One of them got killed by the infection, but I'm convinced the other one would've made it if some *twat* didn't buff dragonflies to see farther than their wingspan (whoops.). 

From a general game balance/design perspective I think balancing an effect should happen based on its usual environment, and for infections that's an early-game character getting bitten and waiting 8-12 hours for it to progress. At that point you should have scouted enough of the map to have a safe-ish location you can ride it out, and you get a good few hours of frenzied searching to hope for an antibiotic.

**Comment by pushmain** [comment]
> most studies land on ~25% with such fancy-schmancy things as "modern medicine", "ICUs" and "clean ORs".
Checked first link. 25% in 24 hours period, yes. Tells nothing. Besides, based on 60 patients treatments only.
The first research results I have provided are based on 15,000 recorded cases of sepsis in Ireland. In it the lifetime expectancy is 12 hours. The other is similar.
Point is, in-game survival probability of infected gets huge hit in this scenario.
And from what I saw, most likely in other scenarios also. To the point of breaking the game balance, which was fine.

**Comment by Venera3** [comment]
Infections cooling you below 0K are an issue, yes. Infections only influencing you in a relevant way in their last stage is a bit more debatable, but since in normal gameplay you have ample warning and a previous effect you could cure it's fair enough imho.
You still have much better chances than you would in real life, which is compensated by the fact that the whole timeline of the bite->infection progression is fast-forwarded ( though cleaning a wound should be more involved, so it swings around in the player's favor again).
Starting with a malus that is supposed to make a semi-established character go "oh sh*t" should have a low survival chance. Nobody complains about Fungal Infected being unfair, for reference.

**Comment by pushmain** [comment]
> Nobody complains about Fungal Infected being unfair, for reference.

It works OK, as was the infection timeline. Make fungal infection several times faster to kill and there will be complains.

**Comment by adamkad1** [comment]
I thought we were going to realism?

> > The cause is not okay. No infection incapacitate in 3 hours.
> 
> First thing, challenges are meant to be challenging. Then you don't know specifics of infection too: what exact infection it is, how long since it settled in your body, etc.

i thought we were pushing for realism above all else?

**Comment by adamkad1** [comment]
I wish npcs could feed you and give you drink if you are badly infected but recovering (funny thing is, i got a badly infected wound before it started recovering, my tiredness went up so high during the 'badly infected (recovering)' part that it was several times past 'randomly fall asleep threshold'. also i had antibiotics and infection resistant, i was just unlucky

**Comment by pushmain** [comment]
>  also i had good antibiotics and infection resistant, i was just unlucky i guess. 
You already had good antibiotics. Once you no stamina, you have no chances no find them.
Getting green infection is pretty easy, no matter how careful you are.
Disinfectants may be not available, for example when you start in wilderness or in small settlement.
You may be blocked by horde of 50 child zombies.  
When you have to use fire to clean the wound and it works about 50/50.

**Comment by Venera3** [comment]
>my tiredness went up so high during the badly infected (or was it pus filled? idk) '(recovering)' part that it was several times past 'randomly fall asleep threshold'.

Sleepiness gain for infection resistant characters is capped at just over "Tired" (250) and  about halfway to Tired (125) for the infection itself and for recovery respectively. If you fell asleep spontaneously you were playing before the changes.

>Dead in 2 hours, 3 at most. Very realistic and funny.

You vastly overstate its effects. For posterity, the maluses for a non-resistant character are for the 3 stages:
Stamina regen comes out to 11/s, 8/s, and 0/s  (down from 20/s, walking eats 15 stamina a tile, max 10,000);
Thirst is raised by about 24/h including base need (max and death at 1200);
Sleepiness gain is ~190/h, ~360/h and 700/h, (Exhausted begins at 600 with chances of dozing off, capped to 700);
Vomit chance scales from 1/45h to 1/5h, averaging out one-two throwups in an infection;
Speed malus -10, -20 and -30.

So yes, the thing that is life-threatening in real life is life-threatening in game. You still have a good few hours to hope to find antibiotics, but you can't just ignore it until it hits Pus-filled anymore. If you don't want to start with a deadly debuff, don't choose a scenario that stacks the deck against you.

**Comment by pushmain** [comment]
> So yes, the thing that is life-threatening in real life is life-threatening in game.
Not convincing. No real life data behind 2-3 hours of day activity tops.
In-game, more feedback needed, I guess. From other scenarios.
Still think its broken.

**Comment by Venera3** [comment]
Septic patients don't stand up, if they manage to stay conscious at all.  Most that manage that can do so because of intense vasopressor and iv fluid therapy, the latter of which you have no access to. The former is in game with the generic stimulants, especially with an adrenaline shot.

**Comment by pushmain** [comment]
Don't think so.
Its flu by all initial symptoms. Flu patients wont get that kind of treatments.
So the patient gets his intensive care with first antibiotics.

**Comment by pushmain** [comment]
"Within possibly six hours you are going to be feeling incredibly sick." For me, its 6 walking hours.

**Comment by Zireael07** [comment]
As @ZhilkinSerg and @Venera3 told you, it's an unspecified infection - it is most certainly NOT a simple flu.

**Comment by pushmain** [comment]
It was about why RL septic patients wont get iv at once.

> The symptoms of sepsis mimic those of the flu; high temperature, rapid heart rate, rapid breathing, pain, pale or mottled skin, and generally feeling very sick.
> The main difference between sepsis symptoms and flu is sepsis will come on very quickly whereas flu comes on over days

**Comment by Venera3** [comment]
Establishing a high-bore iv line and fluid rescuscitation with at least 30ml/kg is *literally* the first non-diagnostic step in sepsis treatment. That's your rule of thumb dosage for the first three hours, but fluid rescue is continued until lactate levels normalise, vasopressor requirements begin to fall or when fluid need is confirmed to have been met (measured by caval ultrasound or clinical exam). 

Until we have iv fluids and a more robust vitals simulation (we're gonna need both for the wound rework), you can get away with a bit of thirst and needing a nap or five.

https://link.springer.com/article/10.1007/s00134-018-5085-0 , the '18 revision of the Surviving Sepsis Campaign Bundle

**Comment by actual-nh** [comment]
It is possible to have a major wound infection (at least in a limb) without sepsis. Admittedly, it may end with needing to amputate the limb.

**Comment by mlangsdorf** [comment]
The survival rates for battlefield amputations during the US Civil War were around 75%; that was before the introduction of anti-septics and anti-biotics and with somewhat spotty nursing support.

So even in the Really Bad Day scenario, if you can get enough food and water to not starve and dehydrate while your body tries to shake the infection, *and* don't get killed by zombies, you should have a reasonable chance of survival even without antibiotics.

That said, pushmain's original report of how infection works in game sounds pretty reasonable.  You might not die from infection if you have food and water, but you're also not getting up and fighting your way through a dozen zombies to raid the grocery store.  If infection is causing so much weariness that you stay in bed all that, that doesn't seem like a broken subsystem in game.  That sounds like the game is reflecting reality.

**Comment by actual-nh** [comment]
> The survival rates for battlefield amputations during the US Civil War were around 75%; that was before the introduction of anti-septics and anti-biotics and with somewhat spotty nursing support.
>
> So even in the Really Bad Day scenario, if you can get enough food and water to not starve and dehydrate while your body tries to shake the infection, _and_ don't get killed by zombies, you should have a reasonable chance of survival even without antibiotics.

I'm not quite sure how the (quite interesting) Civil War 75% amputation survival rate indicates how well someone could fight off infection _without amputation_ (or an otherwise permanently nonfunctional limb).

**Comment by pushmain** [comment]
Let me be more specific about how much the change affects survival in game. Again, that scenario only.

Before the change, without antibiotics character 100% dies in the sleep during first night.
He can walk and fight, sort of, after periods of blackouts, instead of giving up to sleeping.
Total time he had is about 9 a.m. - 3 a.m, fighting time is until about dark,  ~12 hours.
After that, it had been similar to having no Stamina regen, one single zombie will kill the character in a melee.

Its all well known because its far from unusual not getting antibiotics in a day in this scenario.  
So, it is 4 times less time now. 

In other scenarios getting infection in first day likely have same effect now. Not sure about it.
Feedback may be slow, since green infection condition for new character is not everyday condition in regular scenarios.

**Comment by Venera3** [comment]
Balancing an effect should not be based on a challenge scenario starting you with what is intended as a debilitating malus you have ample warning about and can prevent in normal play. I fail to see an optional challenge becoming four/six/whatever times as hard as being "broken".

**Comment by pushmain** [comment]
The scenario have been in the game 8 years at least. Its a part of it. Dismissing the integral part of game as "optional" can not hide the fact of broken balance.
This scenario simply brings this to the surface.

**Comment by Zireael07** [comment]
The fact that the scenario has been in game for a long time does NOT change the fact that it is a CHALLENGE scenario. It is expected that you are incapacitated by the infection.
PS. "Ample warnings" refers to getting an infection in normal gameplay, NOT part of the challenge scenario.

**Comment by pushmain** [comment]
The fact that the scenario is a CHALLENGE scenario does NOT change the fact that it should remains balanced.

**Comment by Zireael07** [comment]
No, challenge scenario does not have to be balanced (especially as you seem to equate balanced with 'roughly as easy as a normal scenario is')

**Comment by pushmain** [comment]
Yes, challenge scenario as integral part have to remain balanced. And no, you have guessed wrong, challenges are accepted whatever they are.

**Comment by pushmain** [comment]
Just let go that "challenge scenario" part already. 
Normal game should remains balanced. It won't.

**Comment by anothersimulacrum** [comment]
Tangentially related to this,but going off of

> The infection as currently implemented is a soft tissue infection with secondary generalization, with the "bite" effect representing the local process. Those are up there on the severity totem pole, since they usually involve multiple species and have a reservoir in the wound itself that the immune system obviously can't clear on its own. Ideally you'd need to clean the wound to have any chance of survival at all, but revamping them wholesale is a project for later.

It sounds like we may need some more nuance/expansion on the local process.

I don't think the behaviour reported there (I started the game with sepsis and it's really hard) is incorrect, but going from a bite you can clean away instantly with antiseptic to a full-body infection seems a little abbreviated at best, and we could use something like a `local_infection` or whatever effect that turns into a full-body infection after some time.

**Comment by pushmain** [comment]
> going from a bite you can clean away instantly with antiseptic to a full-body infection
 
 The process correctly has 3 stages. What could be added? Inflammation that can be treated with aspirin?

**Comment by Zireael07** [comment]
Inflammation and/or a general "I feel achy and I wonder whether it has anything to do with that bite" stage would be a good thing to add (yes, with some pain but no other effects)

**Comment by pushmain** [comment]
Agree, this stage is missed. Real warnings.

**Comment by Venera3** [comment]
Both bites and infections tick up in duration unless you succeed in a hidden recovery roll.
Bites progress to infection in 6 hours barring repeat bites to the limb ( 40 minutes extra duration) or botched cauterization, starting to hurt at the 30 minute mark and increasing in effect ( pain becomes more common and you get a mild -1/-2 Dexterity debuff ) after an hour. Infection kills you instantly when its duration reaches 24 hours. Both of these systems are unchanged.

This means that in normal play you have 6 hours to find antiseptic, hidden in such obscure locations as "every third bathroom", while being at most mildly inconvenienced by the local process. Those 6 hours, the shiny blue limb indicator, the @ menu and the periodic "your limb hurts!" fit the bill of real warnings in my head, but I may be too attentive for my own good.

The progression timeline being accelerated *is* something I consider unnecessary, but overhauling the hardcoded parts is going to be a bigger project. In normal play it's balanced out by wound treatment being easily accessible, foolproof, practically instant, painless and 100% reliable. At present the 24h timer limits how much scaling gets to happen, but the previous timeline of "I'm fine for 16 hours and then drop dead" is more jarring than the alternative.

**Comment by actual-nh** [comment]
Is a recovery roll failure required for a bite to result in infection?

**Comment by anothersimulacrum** [comment]
> The progression timeline being accelerated is something I consider unnecessary, but overhauling the hardcoded parts is going to be a bigger project. In normal play it's balanced out by wound treatment being easily accessible, foolproof, practically instant, painless and 100% reliable. At present the 24h timer limits how much scaling gets to happen, but the previous timeline of "I'm fine for 16 hours and then drop dead" is more jarring than the alternative.

FWIW, I'm willing to help with whatever C++ is needed here (even if I have to deal with effects, yuck!).

**Comment by pushmain** [comment]
Challenge - Really Bad Day Scenario remains playable.

**Comment by pushmain** [comment]
Since some changes are planned, lets keep the issue open.

**Comment by pushmain** [comment]
Freezing bug likely has been properly fixed by https://github.com/CleverRaven/Cataclysm-DDA/pull/47411
